src/models/appmodel.ts

```diff
@@ -11,17 +12,39 @@ export class AppModel implements IAppModel {
 
     @observable private _selectedFlavor = "";
     get selectedFlavor(): string { return this._selectedFlavor; }
-    set selectedFlavor(item: string) { this._selectedFlavor = item; }
+    set selectedFlavor(item: string) { 
+        this._selectedFlavor = item; 
+        if(!this._isConstructing) this.saveState();
+    }
 
+    private _isConstructing = true;q
     @observable private _textInput = "";
     get flavorInput(): string { return this._textInput; }
     set flavorInput(value: string) { this._textInput = value; }
 
     get flavorTextIsValid() { return Boolean(this.flavorInput); }
 
-    constructor()
+    private _dataModel: IDataModel;
+
+    constructor(dataModel: IDataModel)
     {
         this.selectedFlavor = this.flavors[0];
+        this._dataModel = dataModel;
+        const data = dataModel.load();
+        if(data != null && data != "")
+        {
+            this.flavors.clear();
+            const jsonData = JSON.parse(data);
+            this.selectedFlavor = jsonData.selected;
+            jsonData.flavors.forEach((f: any) => {
+                this.flavors.push(f);
+            });
+            if(this.selectedFlavor == null || this.selectedFlavor === 'undefined') {
+                this.selectedFlavor = jsonData.flavors[0];
+            }
+
+        }
+        this._isConstructing = false;
```

Comment:
I'm not sure that is important.
You can use the simple checking in the `this.saveState` method if the storage instance available

Reply:
Yeah, that makes more sense.